File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public class Hamming
{
    public static int Compute(string strand1, string strand2)
    {
        return Math.Abs(strand1.Length - strand2.Length) + strand1.Where((t, i) => t != strand2[i]).Count();
    }
}

File: swift/hamming/Hamming.swift
import Foundation

class Hamming {
    class func compute(_ strand : String, against strand2: String) -> Int?  {
        guard (strand.characters.count == strand2.characters.count) else {
            return nil;
        }
        return zip(strand.characters, strand2.characters).filter(!=).count
    }
}
